0.3.30: darwin x86_64 failure in `smin_negative`

On macos x86_64 15.7 sdk26/clang17, there is a failure in the `smin_negative` test:

```
TEST 1/125 min:smin_negative make[1]: *** [Makefile:86: run_test] Bus error: 10
```

```
$ lldb -- ./openblas_utest
(lldb) run
Process 82549 launched: '/var/folders/fy/f3nz745x07j5nkpmrh_0jlbm0000gq/T/hin601/spack-stage/spack-stage-openblas-0.3.30-ouflo6blmevfyh2bb3gqecxudkst6lrb/spack-src/utest/openblas_utest' (x86_64)
TEST 1/125 min:smin_negative Process 82549 stopped
* thread #1, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=2, address=0x1019b8300)
    frame #0: 0x000000010003e130 openblas_utest`.L998 + 32
openblas_utest`.L998:
->  0x10003e130 <+32>: addb   %al, (%rax)
    0x10003e132 <+34>: addb   %al, (%rax)
    0x10003e134 <+36>: addb   %al, (%rax)
    0x10003e136 <+38>: addb   %al, (%rax)
Target 0: (openblas_utest) stopped.
(lldb) bt
* thread #1, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=2, address=0x1019b8300)
  * frame #0: 0x000000010003e130 openblas_utest`.L998 + 32
    frame #1: 0x0000000100002e9a openblas_utest`__ctest_min_smin_negative_run + 74
    frame #2: 0x0000000100002a4c openblas_utest`ctest_main + 1100
    frame #3: 0x00007ff81c802530 dyld`start + 3056
(lldb) disassemble -f
openblas_utest`.L998:
    0x10003e110 <+0>:  minps  %xmm1, %xmm0
    0x10003e113 <+3>:  minps  %xmm3, %xmm2
    0x10003e116 <+6>:  minps  %xmm2, %xmm0
    0x10003e119 <+9>:  movaps %xmm0, %xmm1
    0x10003e11c <+12>: movhlps %xmm0, %xmm0 ; xmm0 = xmm0[1,1] 
    0x10003e11f <+15>: minps  %xmm1, %xmm0
    0x10003e122 <+18>: movaps %xmm0, %xmm1
    0x10003e125 <+21>: shufps $0x1, %xmm0, %xmm0 ; xmm0 = xmm0[1,0,0,0] 
    0x10003e129 <+25>: minss  %xmm1, %xmm0
    0x10003e12d <+29>: nopl   (%rax)
->  0x10003e130 <+32>: addb   %al, (%rax)
    0x10003e132 <+34>: addb   %al, (%rax)
    0x10003e134 <+36>: addb   %al, (%rax)
    0x10003e136 <+38>: addb   %al, (%rax)
    0x10003e138 <+40>: addb   %al, (%rax)
    0x10003e13a <+42>: addb   %al, (%rax)
    0x10003e13c <+44>: addb   %al, (%rax)
    0x10003e13e <+46>: addb   %al, (%rax)
```

I'm a bit iffy on this, but it has the feeling of an alignment issue? 
This failure does not occur on aarch64 macos.

--- a/kernel/x86_64/iamax_sse.S
+++ b/kernel/x86_64/iamax_sse.S
@@ -96,9 +96,9 @@
 #ifdef USE_ABS
 	andps	%xmm15, %xmm0
 #endif
-	movaps	%xmm0, %xmm1
-	movaps	%xmm0, %xmm2
-	movaps	%xmm0, %xmm3		/* Generating "seed value" */
+	movups	%xmm0, %xmm1
+	movups	%xmm0, %xmm2
+	movups	%xmm0, %xmm3		/* Generating "seed value" */
 	cmpq	$SIZE, INCX
 	jne	.L80			/* Incx != 1 goto L80 */
 
@@ -146,25 +146,25 @@
 	PREFETCH (PREFETCHSIZE +  0) - PREOFFSET(X)
 #endif
 
-	movaps	 0 * SIZE(X), %xmm4
+	movups	 0 * SIZE(X), %xmm4
 #ifdef USE_ABS
 	andps	%xmm15, %xmm4
 #endif
 	MAXPS	%xmm4, %xmm0
 
-	movaps	 4 * SIZE(X), %xmm5
+	movups	 4 * SIZE(X), %xmm5
 #ifdef USE_ABS
 	andps	%xmm15, %xmm5
 #endif
 	MAXPS	%xmm5, %xmm1
 
-	movaps	 8 * SIZE(X), %xmm6
+	movups	 8 * SIZE(X), %xmm6
 #ifdef USE_ABS
 	andps	%xmm15, %xmm6
 #endif
 	MAXPS	%xmm6, %xmm2
 
-	movaps	12 * SIZE(X), %xmm7
+	movups	12 * SIZE(X), %xmm7
 #ifdef USE_ABS
 	andps	%xmm15, %xmm7
 #endif
@@ -182,13 +182,13 @@
 	testq	$8, M
 	je	.L16
 
-	movaps	0 * SIZE(X), %xmm4
+	movups	0 * SIZE(X), %xmm4
 #ifdef USE_ABS
 	andps	%xmm15, %xmm4
 #endif
 	MAXPS	%xmm4, %xmm0
 
-	movaps	4 * SIZE(X), %xmm5
+	movups	4 * SIZE(X), %xmm5
 #ifdef USE_ABS
 	andps	%xmm15, %xmm5
 #endif
@@ -200,7 +200,7 @@
 	testq	$4, M
 	je	.L17
 
-	movaps	0 * SIZE(X), %xmm6
+	movups	0 * SIZE(X), %xmm6
 #ifdef USE_ABS
 	andps	%xmm15, %xmm6
 #endif
@@ -238,10 +238,10 @@
 	MAXPS	%xmm1, %xmm0
 	MAXPS	%xmm3, %xmm2
 	MAXPS	%xmm2, %xmm0
-	movaps	%xmm0, %xmm1
+	movups	%xmm0, %xmm1
 	movhlps %xmm0, %xmm0
 	MAXPS	%xmm1, %xmm0
-	movaps	%xmm0, %xmm1
+	movups	%xmm0, %xmm1
 	shufps	$1, %xmm0, %xmm0
 	MAXSS	%xmm1, %xmm0
 	shufps	$0, %xmm0, %xmm0
@@ -295,13 +295,13 @@
 	PREFETCH (PREFETCHSIZE +  0) - PREOFFSET(X)
 #endif
 
-	movaps	0 * SIZE(X), %xmm1
+	movups	0 * SIZE(X), %xmm1
 #ifdef USE_ABS
 	andps	 %xmm15, %xmm1
 #endif
 	cmpeqps	%xmm0, %xmm1
 
-	movaps	4 * SIZE(X), %xmm3
+	movups	4 * SIZE(X), %xmm3
 #ifdef USE_ABS
 	andps	 %xmm15, %xmm3
 #endif
@@ -536,10 +536,10 @@
 	MAXPS	%xmm1, %xmm0
 	MAXPS	%xmm3, %xmm2
 	MAXPS	%xmm2, %xmm0
-	movaps	%xmm0, %xmm1
+	movups	%xmm0, %xmm1
 	movhlps %xmm0, %xmm0
 	MAXPS	%xmm1, %xmm0
-	movaps	%xmm0, %xmm1
+	movups	%xmm0, %xmm1
 	shufps	$1, %xmm0, %xmm0
 	MAXSS	%xmm1, %xmm0
 	shufps	$0, %xmm0, %xmm0

